### PR TITLE
feat(chat): Highlight own group mentions with primary

### LINKED ISSUE #9073: Render group mentions primary when the user is a member of the group
Remaining frontend task from https://github.com/nextcloud/spreed/issues/6339

### PR DESCRIPTION
**Pull request by nickvergessen**
### ☑️ Resolves

* Fix #9073 

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
grafik | grafik

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue [hunk 1]
```diff
@@ -27,7 +27,8 @@
 			:primary="true" />
 		<NcUserBubble v-else-if="isGroupMention"
 			:display-name="name"
-			:avatar-image="'icon-group-forced-white'" />
+			:avatar-image="'icon-group-forced-white'"
```
Comment (ShGKme) [inline src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue:30]: **Note:** It's better to not bind a static string. In Vue 3 it works faster when it knows that a prop is not being changed.

`avatar-image="icon-group-forced-white"`

**Review by ShGKme — APPROVED** [review]
Tested, works fine, code is reviewed

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue (hunk 1 above)
Comment (nickvergessen) [reply src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue:30]: Ah, nice to know. But I guess we accidentally do that in quite some places (29 occurrences in 15 files)

File: lib/TInitialState.php [hunk 2]
```diff
@@ -124,6 +126,12 @@ protected function publishInitialStateForUser(IUser $user, IRootFolder $rootFold
 			$this->serverConfig->getUserValue($user->getUID(), 'spreed', 'play_sounds', 'yes') === 'yes'
 		);
 
+
+		$this->initialState->provideInitialState(
+			'user_group_ids',
+			$this->groupManager->getUserGroupIds($user)
+		);
```
Comment (SystemKeeper) [inline lib/TInitialState.php:133]: Is there a way to get these information on clients as well?

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue (hunk 1 above)
Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue:30]: This is kind of a micro-optimization actually. Like "one more comparison during re-rendering".

But there was a rule for this even in Vue 2 for better readability 

https://eslint.vuejs.org/rules/no-useless-v-bind.html#vue-no-useless-v-bind

Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue:30]: This is kind of a micro-optimization actually. Like "one more comparison during re-rendering".

But there was a rule for this even in Vue 2 for better readability 

https://eslint.vuejs.org/rules/no-useless-v-bind.html#vue-no-useless-v-bind

File: lib/TInitialState.php (hunk 2 above)
Comment (nickvergessen) [reply lib/TInitialState.php:133]: I think the URL is

`GET /ocs/v2.php/cloud/users/{userid}/groups`

Comment (ShGKme) [reply lib/TInitialState.php:133]: > Is there a way to get these information on clients as well?

This is also a part of userMetadata, so you already request it with other user info

Comment (SystemKeeper) [reply lib/TInitialState.php:133]: Maybe we just don’t store it yet 🧐 I’ll take a look, thanks to both!

File: src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue (hunk 1 above)
Comment (ShGKme) [reply src/components/MessagesList/MessagesGroup/Message/MessagePart/Mention.vue:30]: At least now Vue 3 handles it automatically
[proof-link](https://template-explorer.vuejs.org/#eyJzcmMiOiI8ZGl2PlxyXG4gIDxkaXYgZGF0YS10ZXN0PVwic3RyaW5nXCI+RGl2IHdpdGggc3RhdGljIHZhbHVlPC9kaXY+XHJcbiAgPGRpdiA6ZGF0YS10ZXN0PVwiJ2pzLXN0cmluZydcIj5EaXYgd2l0aCBiaW5kaW5nIHN0YXRpYyB2YWx1ZTwvZGl2PlxyXG4gIDxkaXYgOmRhdGEtdGVzdD1cInZhcmlhYmxlXCI+RGl2IHdpdGggYmluZGluZyBkeW5hbWljIHZhbHVlPC9kaXY+XHJcbjwvZGl2PiIsIm9wdGlvbnMiOnt9fQ==)

**Pull request merged**